### PR TITLE
Use develop branches for repositories in ufs bundle

### LINKED ISSUE #29: Update JEDI code in ufs-bundle to run with crtm@2.4.1 (or v3)
Currently the ufs-bundle uses hashes of the JEDI repositories that don't work with `crtm@2.4.1-jedi`, which is the only option in spack-stack-1.4.0.

As a workaround, I am installing `crtm@2.4-jedi.2` on AWS parallelcluster in spack-stack-1.4.0, but we need to update ufs-bundle to use newer JEDI hashes that work with `crtm@2.4.1-jedi`.

### PR DESCRIPTION
**Pull request by cmgas**
## Description
Fixes #29 
Working on develop would be easier to maintain (for me, at least!)

Goes with:
- [x] https://github.com/JCSDA-internal/fv3-jedi/pull/950
- [x] https://github.com/JCSDA-internal/soca/pull/918
- [x] https://github.com/JCSDA-internal/fv3-jedi-data/pull/77

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -37,32 +37,28 @@ endif()
...
 # ECMWF libs
 # ----------
-option("BUNDLE_SKIP_ECKIT" "Don't build eckit" "ON" ) # Skip eckit build unless user passes -DBUNDLE_SKIP_ECKIT=OFF
-option("BUNDLE_SKIP_FCKIT" "Don't build fckit" "ON") # Skip fckit build unless user passes -DBUNDLE_SKIP_FCKIT=OFF
-option("BUNDLE_SKIP_ATLAS" "Don't build atlas" "ON") # Skip atlas build unless user passes -DBUNDLE_SKIP_ATLAS=OFF
-
-ecbuild_bundle( PROJECT eckit GIT "https://github.com/ecmwf/eckit.git" TAG 1.20.2 )
-ecbuild_bundle( PROJECT fckit GIT "https://github.com/ecmwf/fckit.git" TAG 0.10.0 )
-ecbuild_bundle( PROJECT atlas GIT "https://github.com/ecmwf/atlas.git" TAG 0.32.1 )
+# ecbuild_bundle( PROJECT eckit GIT "https://github.com/ecmwf/eckit.git" TAG 1.20.2 )
```
Comment (climbfuji) [inline CMakeLists.txt]:
```suggestion
# ecbuild_bundle( PROJECT eckit GIT "https://github.com/ecmwf/eckit.git" TAG 1.23.0 )
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -37,32 +37,28 @@ endif()
...
+# ecbuild_bundle( PROJECT fckit GIT "https://github.com/ecmwf/fckit.git" TAG 0.10.0 )
```
Comment (climbfuji) [inline CMakeLists.txt]:
```suggestion
# ecbuild_bundle( PROJECT fckit GIT "https://github.com/ecmwf/fckit.git" TAG 0.10.1 )
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -37,32 +37,28 @@ endif()
...
+# ecbuild_bundle( PROJECT atlas GIT "https://github.com/ecmwf/atlas.git" TAG 0.32.1 )
```
Comment (climbfuji) [inline CMakeLists.txt]:
```suggestion
# ecbuild_bundle( PROJECT atlas GIT "https://github.com/ecmwf/atlas.git" TAG 0.33.0 )
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -37,32 +37,28 @@ endif()
...
+ecbuild_bundle( PROJECT saber GIT "https://github.com/jcsda-internal/saber.git" BRANCH develop UPDATE )
 
 
 if(UFS_APP MATCHES "^(ATMAERO)$" OR UFS_APP MATCHES "^(S2S)$" OR UFS_APP MATCHES "^(NG-GODAS)$")
-  ecbuild_bundle( PROJECT gsw GIT "https://github.com/jcsda-internal/GSW-Fortran.git" TAG 1a02ebaf6f7a4e9f2c2d2dd973fb050e697bcc74 )  # develop on apr 18
+  ecbuild_bundle( PROJECT gsw GIT "https://github.com/jcsda-internal/GSW-Fortran.git" BRANCH develop UPDATE )
 endif()
 
-#ecbuild_bundle( PROJECT crtm  GIT "https://github.com/jcsda-internal/crtm.git"  TAG v2.4-jedi.2 )
+ecbuild_bundle( PROJECT crtm  GIT "https://github.com/jcsda-internal/crtm.git"  TAG v2.4.1-jedi )
```
Comment (climbfuji) [inline CMakeLists.txt]:
```suggestion
ecbuild_bundle( PROJECT crtm GIT "https://github.com/JCSDA/CRTMv3.git" BRANCH develop UPDATE )
```

**Review by climbfuji — COMMENTED** [review]
@shlyaeva @fcvdb I just pushed an update to the top-level `README.md` to this branch, commit https://github.com/JCSDA/ufs-bundle/pull/30/commits/9e4a5762327fa1137446c548e477dd84c7f9f83d.

Can you please review the combined changes and my additional suggestions for change as part of this review? I am currently testing the PR on my macOS.

**Review by climbfuji — APPROVED** [review]
This all works for me, I can compile UFS_APPs ATM, ATMAERO, NG-GODAS, S2S. The ATM ctests (warmstart) pass, the coldstart tests fail as expected.

File: README.md [hunk 5]
```diff
@@ -9,37 +9,40 @@ This bundle requires the following spack-stack modules be loaded (all except `fm
...
+On a pre-configured platform that has spack-stack-1.4.1 (or 1.4.0) installed, load the spack-stack modules as described in https://spack-stack.readthedocs.io/en/1.4.1 (https://spack-stack.readthedocs.io/en/1.4.0), followed by `module load jedi-ufs-env`. On a configurable (user) platform, follow the instructions in https://spack-stack.readthedocs.io/en/1.4.1 to build the `skylab-dev` or `unified-dev` environment, then run (use the appropriate path, compiler/mpi/python versions for your system):
+```
+module use /Users/heinzell/prod/spack-stack-v1/envs/unified-dev/install/modulefiles/Core
 module load stack-apple-clang/13.1.6
 module load stack-openmpi/4.1.5
 module load stack-python/3.10.8
-module load jedi-ufs-env/skylab-dev
+module load jedi-ufs-env/unified-dev
+```
+Further, for testing of the newest `fms@2023.02-beta1` tag, verify if the platform you are on has this version available and if yes, run
```
Comment (shlyaeva) [inline README.md:42]: Does this include the change that soca needed?

**Review by shlyaeva — APPROVED** [review]
Looks good to me, and as @climbfuji suggested, would be great to have CI on this now that it uses develop for most branches in the bundle (otherwise it may be hard to find which changes will be breaking it eventually).

File: README.md (hunk 5 above)
Comment (climbfuji) [reply README.md:42]: Yes!

Comment (climbfuji) [reply README.md:42]: @travissluka tested this already (he said)

Comment (shlyaeva) [reply README.md:42]: 🎉  you rock!
we'll test with fv3-jedi too, unless @travissluka you already tested with both soca and fv3-jedi?

Comment (climbfuji) [reply README.md:42]: Well, the tag of fms contains the changes that SOCA needs, but the ufs-weather-model, fv3-jedi-linearmodel and fv3-jedi don't work with that tag ye. See https://github.com/ufs-community/ccpp-physics/issues/90 and https://github.com/JCSDA-internal/fv3-jedi-linearmodel/issues/18 for examples. Also, see https://github.com/NOAA-GFDL/FMS/issues/1284 for bugs in the build system of that tag.

Comment (shlyaeva) [reply README.md:42]: Got it, so testing https://github.com/JCSDA-internal/fv3-bundle/pull/103 (with updates to jcsda repos) isn't expected to work?

Comment (climbfuji) [reply README.md:42]: At least at the moment, not. Let's see how far we get this week.

**Pull request merged**